File: internal/models/game_node.go
package models

import "time"

// GameNodeType 游戏节点类型
type GameNodeType string

const (
	GameNodeTypePhysical GameNodeType = "physical" // 物理节点
	GameNodeTypeVirtual  GameNodeType = "virtual"  // 虚拟节点
)

// GameNodeState 游戏节点状态
type GameNodeState string

const (
	GameNodeStateOffline GameNodeState = "offline" // 离线
	GameNodeStateOnline  GameNodeState = "online"  // 在线
	GameNodeStateReady   GameNodeState = "ready"   // 就绪
	GameNodeStateBusy    GameNodeState = "busy"    // 忙碌
	GameNodeStateError   GameNodeState = "error"   // 错误
)

// GameNodeStatus 节点状态信息
type GameNodeStatus struct {
	State      GameNodeState          `json:"state" yaml:"state"`             // 节点状态
	Online     bool                   `json:"online" yaml:"online"`           // 是否在线
	LastOnline time.Time              `json:"last_online" yaml:"last_online"` // 最后在线时间
	UpdatedAt  time.Time              `json:"updated_at" yaml:"updated_at"`   // 状态更新时间
	Resources  map[string]string      `json:"resources" yaml:"resources"`     // 资源使用情况
	Metrics    map[string]interface{} `json:"metrics" yaml:"metrics"`         // 性能指标
}

// GameNode 游戏节点
type GameNode struct {
	ID        string            `json:"id" yaml:"id"`                 // 节点ID
	Name      string            `json:"name" yaml:"name"`             // 节点名称
	Model     string            `json:"model" yaml:"model"`           // 节点型号
	Type      GameNodeType      `json:"type" yaml:"type"`             // 节点类型
	Location  string            `json:"location" yaml:"location"`     // 节点位置
	Hardware  map[string]string `json:"hardware" yaml:"hardware"`     // 硬件配置
	Network   map[string]string `json:"network" yaml:"network"`       // 网络配置
	Labels    map[string]string `json:"labels" yaml:"labels"`         // 标签
	Status    GameNodeStatus    `json:"status" yaml:"status"`         // 节点状态信息
	CreatedAt time.Time         `json:"created_at" yaml:"created_at"` // 创建时间
	UpdatedAt time.Time         `json:"updated_at" yaml:"updated_at"` // 更新时间
}

// TableName 返回表名
func (GameNode) TableName() string {
	return "game_nodes"
}

File: frontend/src/types/GameNode.ts
export interface NodeHardware {
  cpu: string;
  memory: string;
  disk: string;
}

export interface NodeNetwork {
  ip: string;
}

export interface Node {
  id: string;
  name: string;
  status: "online" | "offline" | "maintenance";
  hardware: NodeHardware;
  network: NodeNetwork;
  lastHeartbeat: string;
}

export interface NodeMonitor {
  cpu: {
    usage: number;
    temperature: number;
    cores: number;
  };
  memory: {
    total: number;
    used: number;
    free: number;
  };
  disk: {
    total: number;
    used: number;
    free: number;
  };
  network: {
    upload: number;
    download: number;
  };
}

export enum GameNodeType {
  Physical = 'physical',
  Virtual = 'virtual',
  Container = 'container'
}

export enum GameNodeState {
  Offline = 'offline',
  Online = 'online',
  Maintenance = 'maintenance',
  Ready = 'ready',
  Busy = 'busy',
  Error = 'error'
}

export interface CPUHardware {
  model: string;
  cores: number;
  threads: number;
  frequency: number;
  cache: number;
}

export interface MemoryHardware {
  total: number;
  type: string;
  frequency: number;
  channels: number;
}

export interface GPUHardware {
  model: string;
  memoryTotal: number;
  cudaCores: number;
}

export interface StorageDevice {
  type: string;
  capacity: number;
}

export interface StorageHardware {
  devices: StorageDevice[];
}

export interface CPUMetrics {
  usage: number;
  temperature: number;
}

export interface MemoryMetrics {
  available: number;
  used: number;
  usage: number;
}

export interface GPUMetrics {
  usage: number;
  memoryUsed: number;
  memoryFree: number;
  memoryUsage: number;
  temperature: number;
  power: number;
}

export interface StorageMetrics {
  used: number;
  free: number;
  usage: number;
}

export interface NetworkMetrics {
  bandwidth: number;
  latency: number;
  connections: number;
  packetLoss: number;
}

export interface HardwareInfo {
  cpu: CPUHardware;
  memory: MemoryHardware;
  gpu: GPUHardware;
  storage: StorageHardware;
}

export interface MetricsInfo {
  cpu: CPUMetrics;
  memory: MemoryMetrics;
  gpu: GPUMetrics;
  storage: StorageMetrics;
  network: NetworkMetrics;
}

export interface ResourceInfo {
  id: string;
  timestamp: number;
  hardware: HardwareInfo;
  metrics: MetricsInfo;
}

export interface Metric {
  name: string;
  type: string;
  value: number;
  labels: Record<string, string>;
}

export interface MetricsReport {
  id: string;
  timestamp: number;
  metrics: Metric[];
}

export interface GameNodeStatus {
  state: GameNodeState;
  online: boolean;
  lastOnline: string;
  updatedAt: string;
  resource: ResourceInfo;
  metrics: MetricsReport;
}

export interface GameNode {
  id: string;
  alias: string;
  model: string;
  type: GameNodeType;
  location: string;
  labels: Record<string, string>;
  hardware: Record<string, string>;
  system: Record<string, string>;
  status: GameNodeStatus;
  createdAt: string;
  updatedAt: string;
}

export interface GameNodeQuery {
  page: number;
  pageSize: number;
  keyword?: string;
  status?: string;
}

export type GameNodeForm = Partial<GameNode>;

export interface GameNodeListParams {
  page?: number;
  size?: number;
  keyword?: string;
  status?: GameNodeState;
  type?: GameNodeType;
  sortBy?: 'created_at' | 'updated_at' | 'status';
  sortOrder?: 'asc' | 'desc';
}

export interface GameNodeListResult {
  total: number;
  items: GameNode[];
}
